src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/api/CardEventHandlerTest.java

```diff
@@ -0,0 +1,180 @@
+    Round newRound = cardEventHandler.createRound(match, round.getRoundNumber());
+    assertEquals(2, newRound.getRoundNumber());
+    Collection<Round> savedRounds = roundRepository.findAll();
+    assertTrue(savedRounds.stream().anyMatch(r -> r.getRoundNumber() == 2));
+  }
+
+  @Test
+  public void get_number_of_played_cards_one_card() {
+    setUpRelatedEntities();
+    CardEventHandler cardEventHandler = new CardEventHandler(roundRepository, cardRepository);
+    assertEquals(1, cardEventHandler.getNumberOfPlayedCards(round));
+  }
+
+  @Test
+  public void get_number_of_played_cards_no_card() {
+    matchRepository.save(match);
+    round.setMatch(match);
+    round.setRoundNumber(1);
+    roundRepository.save(round);
+    CardEventHandler cardEventHandler = new CardEventHandler(roundRepository, cardRepository);
+    assertEquals(0, cardEventHandler.getNumberOfPlayedCards(round));
+  }
+
+  @Test
+  public void get_number_of_played_cards_multiple_cards() {
+    setUpRelatedEntities();
+    cardRank = CardRank.EIGHT;
+    cardSuit = CardSuit.HEART;
+    Card card2 = new Card(cardRank, cardSuit);
+    card2.setRound(round);
+    cardRepository.save(card2);
+    cardRank = CardRank.TEN;
+    cardSuit = CardSuit.HEART;
+    Card card3 = new Card(cardRank, cardSuit);
+    card3.setRound(round);
+    cardRepository.save(card3);
+    // a fourth round which is not assigned to this round (and therefore shall not be counted)
+    cardRank = CardRank.TEN;
+    cardSuit = CardSuit.SPADE;
+    Card card4 = new Card(cardRank, cardSuit);
+    cardRepository.save(card4);
+    CardEventHandler cardEventHandler = new CardEventHandler(roundRepository, cardRepository);
+    assertEquals(3, cardEventHandler.getNumberOfPlayedCards(round));
+  }
+
+  @Test
+  public void set_old_round_to_inactive()
+      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    setUpRelatedEntities();
+    CardEventHandler cardEventHandler = new CardEventHandler(roundRepository, cardRepository);
+    /*
+    Accessing private Methods for testing:
+    Method privateMethod = CardEventHandler.class.getDeclaredMethod("setOldRoundToInactive", Round.class);
+    privateMethod.setAccessible(true);
+    privateMethod.invoke(cardEventHandler, round);
+     */
+    cardEventHandler.setOldRoundToInactive(round);
+    Collection<Round> savedRounds = roundRepository.findAll();
+    assertTrue(savedRounds.stream().anyMatch(r -> r.isActive() == false));
+    // and no remaining round which is active (as we did not create a new round in this test)
+    assertFalse(savedRounds.stream().anyMatch(r -> r.isActive() == true));
+  }
+
+  /* How do I have to setup the security context correctly to create/save a game?
+  @Test
+  @WithMockUser(username="player1", roles={"ROLE_PLAYER"})
+  public void get_number_of_players() {
+    game.setName("game1");
+    gameRepository.save(game);
```

Comment:
just use gameRepository.saveAll, there is no security check

Reply:
Hab's jetzt ganz ohne Speichern gelöst :)